daemon/daemon.go

```
@@ -679,100 +668,19 @@ func (d *Daemon) init() error {
-	fmt.Fprintf(fw, "#define TUNNEL_ENDPOINT_MAP_SIZE %d\n", tunnel.MaxEntries)
-	fmt.Fprintf(fw, "#define PROXY_MAP_SIZE %d\n", proxymap.MaxEntries)
-	fmt.Fprintf(fw, "#define ENDPOINTS_MAP %s\n", lxcmap.MapName)
-	fmt.Fprintf(fw, "#define ENDPOINTS_MAP_SIZE %d\n", lxcmap.MaxEntries)
-	fmt.Fprintf(fw, "#define METRICS_MAP %s\n", metricsmap.MapName)
-	fmt.Fprintf(fw, "#define METRICS_MAP_SIZE %d\n", metricsmap.MaxEntries)
-	fmt.Fprintf(fw, "#define POLICY_MAP_SIZE %d\n", policymap.MaxEntries)
-	fmt.Fprintf(fw, "#define IPCACHE_MAP %s\n", ipcachemap.Name)
-	fmt.Fprintf(fw, "#define IPCACHE_MAP_SIZE %d\n", ipcachemap.MaxEntries)
-	fmt.Fprintf(fw, "#define POLICY_PROG_MAP_SIZE %d\n", policymap.ProgArrayMaxEntries)
-	fmt.Fprintf(fw, "#define SOCKOPS_MAP_SIZE %d\n", sockmap.MaxEntries)
-
-	if option.Config.DatapathMode == option.DatapathModeIpvlan {
-		fmt.Fprintf(fw, "#define ENABLE_SECCTX_FROM_IPCACHE 1\n")
-	}
-
-	if option.Config.PreAllocateMaps {
-		fmt.Fprintf(fw, "#define PREALLOCATE_MAPS\n")
-	}
-
-	fmt.Fprintf(fw, "#define EVENTS_MAP %s\n", "cilium_events")
-	fmt.Fprintf(fw, "#define POLICY_CALL_MAP %s\n", policymap.CallMapName)
-	fmt.Fprintf(fw, "#define PROXY4_MAP cilium_proxy4\n")
-	fmt.Fprintf(fw, "#define PROXY6_MAP cilium_proxy6\n")
-	fmt.Fprintf(fw, "#define EP_POLICY_MAP %s\n", eppolicymap.MapName)
-	fmt.Fprintf(fw, "#define LB6_REVERSE_NAT_MAP cilium_lb6_reverse_nat\n")
-	fmt.Fprintf(fw, "#define LB6_SERVICES_MAP cilium_lb6_services\n")
-	fmt.Fprintf(fw, "#define LB6_RR_SEQ_MAP cilium_lb6_rr_seq\n")
-	fmt.Fprintf(fw, "#define LB4_REVERSE_NAT_MAP cilium_lb4_reverse_nat\n")
-	fmt.Fprintf(fw, "#define LB4_SERVICES_MAP cilium_lb4_services\n")
-	fmt.Fprintf(fw, "#define LB4_RR_SEQ_MAP cilium_lb4_rr_seq\n")
-
-	fmt.Fprintf(fw, "#define TRACE_PAYLOAD_LEN %dULL\n", option.Config.TracePayloadlen)
-	fmt.Fprintf(fw, "#define MTU %d\n", d.mtuConfig.GetDeviceMTU())
+		fw := bufio.NewWriter(f)
+		defer fw.Flush()
```

Errors were never checked here before, but as a followup we should consider checking the error of the Flush() given that this is a buffered writer.